stablehlo/conversions/linalg/CMakeLists.txt

```diff
@@ -0,0 +1,2 @@
+add_subdirectory(tests)
```

Comment:
Also got a spare minute to write a bazel patch :). Requires rebasing the CL on HEAD first.

```
diff --git a/BUILD.bazel b/BUILD.bazel
index da5071a9..7c9e02e1 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -331,6 +331,62 @@ td_library(
     ],
 )
 
+
+gentbl_cc_library(
+    name = "linalg_pass_inc_gen",
+    strip_include_prefix = ".",
+    tbl_outs = [
+        (
+            [
+                "-gen-pass-decls",
+                "-name=StablehloLinalgTransforms",
+            ],
+            "stablehlo/conversions/linalg/transforms/Passes.h.inc",
+        ),
+    ],
+    tblgen = "@llvm-project//mlir:mlir-tblgen",
+    td_file = "stablehlo/conversions/linalg/transforms/Passes.td",
+    deps = ["@llvm-project//mlir:PassBaseTdFiles"],
+)
+
+cc_library(
+    name = "linalg_passes",
+    srcs = [
+        "stablehlo/conversions/linalg/transforms/LegalizeToLinalgUtils.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloLegalizeToLinalg.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloToArith.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloToLinalgConvolution.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloToLinalgDotProduct.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloToLinalgPointwise.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloToLinalgRandom.cpp",
+        "stablehlo/conversions/linalg/transforms/StableHloToLinalgReduce.cpp",
+        "stablehlo/conversions/linalg/transforms/TypeConversion.cpp",
+    ],
+    hdrs = [
+        "stablehlo/conversions/linalg/transforms/LegalizeToLinalgUtils.h",
+        "stablehlo/conversions/linalg/transforms/MapStableHloToScalarOp.h",
+        "stablehlo/conversions/linalg/transforms/PassDetail.h",
+        "stablehlo/conversions/linalg/transforms/Passes.h",
+        "stablehlo/conversions/linalg/transforms/Rewriters.h",
+        "stablehlo/conversions/linalg/transforms/TypeConversion.h",
+    ],
+    strip_include_prefix = ".",
+    deps = [
+        ":linalg_pass_inc_gen",
+        ":stablehlo_ops",
+        ":chlo_ops",
+        "@llvm-project//mlir:FuncDialect",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:LinalgDialect",
+        "@llvm-project//mlir:LinalgTransforms",
+        "@llvm-project//mlir:LinalgUtils",
+        "@llvm-project//mlir:Pass",
+        "@llvm-project//mlir:Transforms",
+        "@llvm-project//mlir:VectorDialect",
+    ],
+)
+
+
 cc_library(
     name = "reference_axes",
     srcs = [
@@ -948,6 +1004,7 @@ cc_binary(
     ],
     deps = [
         ":interpreter_ops",
+        ":linalg_passes",
         ":register",
         ":stablehlo_passes",
         ":tosa_passes",
@@ -1317,6 +1374,7 @@ test_suite(
     tests = [
         "//stablehlo/tests:stablehlo_tests",
         "//stablehlo/testdata:stablehlo_testdata_tests",
-        "//stablehlo/conversions/tosa/tests:stablehlo_tosa_tests"
+        "//stablehlo/conversions/tosa/tests:stablehlo_tosa_tests",
+        "//stablehlo/conversions/linalg/tests:stablehlo_linalg_tests",
     ],
 )
diff --git a/stablehlo/conversions/linalg/tests/BUILD.bazel b/stablehlo/conversions/linalg/tests/BUILD.bazel
new file mode 100644
index 00000000..25fa3574
--- /dev/null
+++ b/stablehlo/conversions/linalg/tests/BUILD.bazel
@@ -0,0 +1,56 @@
+# Copyright 2023 The StableHLO Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
+load("@llvm-project//llvm:lit_test.bzl", "lit_test", "package_path")
+
+package(
+    default_visibility = ["//visibility:public"],
+    licenses = ["notice"],
+)
+
+# Equivalent of configure_lit_site_cfg from CMakeLists.txt.
+expand_template(
+    name = "lit_site_cfg_py_gen",
+    testonly = True,
+    out = "lit.site.cfg.py",
+    substitutions = {
+        "@LIT_SITE_CFG_IN_HEADER@": "# Autogenerated, do not edit.",
+        "@LLVM_TOOLS_DIR@": package_path("@llvm-project//llvm:BUILD"),
+        "\"@STABLEHLO_TOOLS_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], 'stablehlo')",
+        "\"@STABLEHLO_SOURCE_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], 'stablehlo')",
+    },
+    template = "lit.site.cfg.py.in",
+)
+
+# Equivalent of add_lit_testsuite from CMakeLists.txt.
+[
+    lit_test(
+        name = "%s.test" % src,
+        srcs = [src],
+        data = [
+            "lit.cfg.py",
+            "lit.site.cfg.py",
+            "//:stablehlo-opt",
+            "@llvm-project//llvm:FileCheck",
+        ],
+        size = "small",
+        tags = ["stablehlo_linalg_tests"],
+    )
+    for src in glob(["**/*.mlir"])
+]
+
+test_suite(
+    name = "stablehlo_linalg_tests",
+    tags = ["stablehlo_linalg_tests"],
+)
diff --git a/stablehlo/conversions/linalg/tests/lit.site.cfg.py.in b/stablehlo/conversions/linalg/tests/lit.site.cfg.py.in
index ba683e73..38cf580f 100644
--- a/stablehlo/conversions/linalg/tests/lit.site.cfg.py.in
+++ b/stablehlo/conversions/linalg/tests/lit.site.cfg.py.in
@@ -17,4 +17,4 @@ import lit.llvm
 lit.llvm.initialize(lit_config, config)
 config.llvm_tools_dir = "@LLVM_TOOLS_DIR@"
 config.stablehlo_tools_dir = "@STABLEHLO_TOOLS_DIR@"
-lit_config.load_config(config, "@STABLEHLO_SOURCE_DIR@/stablehlo/conversions/linalg/tests/lit.cfg.py")
+lit_config.load_config(config, "@STABLEHLO_SOURCE_DIR@" + "/stablehlo/conversions/linalg/tests/lit.cfg.py")
```

Reply:
Thanks so much for adding the Bazel build!  Applied patch and spot fixed filenames to match the lowercase "hlo" pattern.